source/io.c

```diff
@@ -350,7 +302,55 @@ static struct aws_error_info s_errors[] = {
     AWS_DEFINE_ERROR_INFO_IO(
         AWS_IO_STREAM_GET_LENGTH_UNSUPPORTED,
         "Get length is not supported in the underlying I/O source."),
+    AWS_DEFINE_ERROR_INFO_IO(
```

Comment:
You could add some defenses against this happening again by adding a test like:
- in io_lib_test.c
- name something like io_library_error_order
- init library, then assert that a few error codes have their current int value. Like maybe pick the first, last, and a middle error code

that would catch someone naively adding an error code to the middle

Reply:
io_library_error_order test added